### PR TITLE
feat: warn when create store with exsist id

### LINKED ISSUE #1394: Warn if two stores are defined with the same name
### What problem is this solving

I have a store. I copied it to make a second store but forgot to change the name of the second store.

This resulted in very subtle bugs where store state that should never be able to be `undefined` was appearing as `undefined`.

It should be noted that these stores are being intentionally used in an un-strict manner where the state is being mutated directly.

**store/invoices.ts**
```typescript
export const useStore = defineStore('invoices', () => {
```
**store/templates.ts**
```typescript
export const useStore = defineStore('invoices', () => {
```

### Proposed solution

Warn if multiple stores share the same name.

### Describe alternatives you've considered

Using the Vue Dev Tools, I was able to see that only one store existed and it sometimes had the values of the first store and sometimes values from the second store.

### PR DESCRIPTION
**Pull request by OXXXRA**
closes https://github.com/vuejs/pinia/issues/1394

### CODE REVIEW

**Review by isbik — APPROVED** [review]
Ok

**Comment by OXXXRA** [comment]
@posva, hi, it's ok?

**Comment by posva** [comment]
Thanks but this seem like a copy of #1452 with only one test added... **That PR implementation doesn't work**, if you try it locally you will see how the console is flushed with warnings when running the tests... Also, the warning is [still not dev only](https://github.com/vuejs/pinia/pull/1452#issuecomment-1217889592).

**Comment by OXXXRA** [comment]
@posva, now i added output only for dev, also checked warn output in playground.
Снимок экрана 2022-08-31 в 20 08 33

File: packages/pinia/__tests__/store.spec.ts [hunk 1]
```diff
@@ -379,4 +379,12 @@ describe('Store', () => {
       `[🍍]: A getter cannot have the same name as another state property. Rename one of them. Found with "anyName" in store "main".`
     ).toHaveBeenWarnedTimes(1)
   })
-})
+  it('warns when create store with exsist id', async () => {
```
Comment (seanohue) [inline packages/pinia/__tests__/store.spec.ts]: This seems like a typo, should be
```suggestion
  it('warns when creating store with existing id', async () => {
```

File: packages/pinia/src/store.ts [hunk 2]
```diff
@@ -866,6 +866,7 @@ export function defineStore(
     options = idOrOptions
     id = idOrOptions.id
   }
+  // push id into stores array
```
Comment (seanohue) [inline packages/pinia/src/store.ts]: This comment is confusing here,  should be on the line before the action its describing, or be removed

Comment (OXXXRA) [reply packages/pinia/src/store.ts]: Thank you, i remove useless comment!

File: packages/pinia/__tests__/store.spec.ts (hunk 1 above)
Comment (OXXXRA) [reply packages/pinia/__tests__/store.spec.ts]: Thank you, i change this text!

**Comment by posva** [comment]
Thanks but this still doesn't work and my feedback is getting ignored so I prefer closing this so other contributors feel free to open a new PR

**Pull request closed without merging**